pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/ClassStats.java

```diff
@@ -159,14 +159,14 @@
      *
      * @return The result of the computation, or {@code Double.NaN} if it couldn't be performed
      */
-    /* default */ double computeWithResultOption(OperationMetricKey key, ASTAnyTypeDeclaration node,
+    /* default */ double computeWithResultOption(MetricKey<OperationMetric> key, ASTAnyTypeDeclaration node,
                                                  boolean force, MetricVersion version, ResultOption option) {
 
         List<ASTMethodOrConstructorDeclaration> ops = findOperations(node, false);
 
         List<Double> values = new ArrayList<>();
         for (ASTMethodOrConstructorDeclaration op : ops) {
-            if (key.supports(op)) {
+            if (key.getCalculator().supports(op)) {
```

Comment:
Maybe it makes sense to define the "supports" method also in the MetricKey interface. Would be a minor improvement. I guess, it would need to be a bit more generic, than it is now, e.g.

    boolean supports(AccessNode node);

It would then be the responsibility of the metric to return false, if the node type is not supported.
Hm... we basically have already a "supports(AccessNode node)" method in AbstractMetric.
Downside is, that each MetricKey implementation needs to delegate one additional method.